crates/core/src/operations/update.rs

```diff
@@ -505,13 +519,32 @@ impl OpManager {
             .ring
             .subscribers_of(key)
             .map(|subs| {
-                subs.value()
+                let all_subs = subs.value();
+                tracing::info!(
+                    "SUBSCRIPTION_DIAG: Contract {} has {} total subscribers",
+                    key.id(),
+                    all_subs.len()
+                );
+                for sub in all_subs.iter() {
```

Comment:
onlyu do this if level debug is enabled

Reply:
Good point! This debug log could be wrapped in a debug level check for performance. Will note for follow-up optimization.